src/Umbraco.Web.UI.Client/gulp/tasks/js.js

```diff
@@ -20,11 +19,10 @@ function js() {
         gulp.src(config.sources.globs.js).pipe( gulp.dest(config.root + config.targets.js) )
     );
 
-    _.forEach(config.sources.js, function (group) {
-        stream.add(
-            processJs(group.files, group.out)
-        );
-    });
+    for(const group in config.sources.js){
+        const groupItem = config.sources.js[group];
+        stream.add(processJs(groupItem.files, groupItem.out));
+    }
```

Comment:
Not sure how true this still is, but it used to be the case at least that `for ... in` looping on an objects keys was not the best option, due to the fact it's prototype chain might include props you don't want to loop. Instead, requesting it's keys and forEeaching should be better.

    Object.keys(config.sources.js).forEach(key => {
         const groupItem = config.sources.js[key];
         stream.add(processJs(groupItem.files, groupItem.out));
    });

Reply:
The power of the HIVE mind